From 5a4472031c06f6acf44c5ef4a778fbdc909fc780 Mon Sep 17 00:00:00 2001
From: Yong Chen <yong.a.chen@intel.com>
Date: Thu, 18 Apr 2013 17:36:30 +0800
Subject: Renderscript: Fix RS test apk problems

BZ: 98426, Fix RS sample display
Multiply rsSample outputs with 255.0f, or else no display for RS sample
Orig-MCG-Change-Id: I22aa7b8f4b94f1af7cf7411f799e2c6696f9791d

BZ: 129663, Fix RS sample no display
Set Allocation.USAGE_GRAPHICS_TEXTURE when preparing RS
Allocation, or else rsSample() returns 0 which causes no
display for RS sample.
Orig-MCG-Change-Id: I8baa55a0e03c5a4d868c082a96b33358576e4226

BZ: 146411, Fix ImageProcessing JB quits when run Intrinsics Conv 3x3
Add Intrinsics Convolve 3x3 in ImageProcessing JB runTest()
Orig-MCG-Change-Id: Idd5c82afd8fa895bfd11865ed245b08da55b42bf

Category: aosp improvement
Domain: AOSP-RenderScript
Origin: internal
Upstream-Candidate: yes

Change-Id: I892b8a9703fafa23c6ecc0cc55acb3f66bc604e0
Signed-off-by: Yong Chen <yong.a.chen@intel.com>
---
 .../src/com/android/rs/image/Convolve3x3.java      |    7 +++++--
 .../com/android/rs/sample/SampleRSActivity.java    |    7 ++++---
 .../SampleTest/src/com/android/rs/sample/sample.rs |    2 +-
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/java/tests/ImageProcessing_jb/src/com/android/rs/image/Convolve3x3.java b/java/tests/ImageProcessing_jb/src/com/android/rs/image/Convolve3x3.java
index d7acf4a..a76ffd2 100644
--- a/java/tests/ImageProcessing_jb/src/com/android/rs/image/Convolve3x3.java
+++ b/java/tests/ImageProcessing_jb/src/com/android/rs/image/Convolve3x3.java
@@ -64,7 +64,10 @@ public class Convolve3x3 extends TestBase {
     }
 
     public void runTest() {
-        mScript.forEach_root(mOutPixelsAllocation);
+        if (mUseIntrinsic) {
+            mIntrinsic.forEach(mOutPixelsAllocation);
+        } else {
+            mScript.forEach_root(mOutPixelsAllocation);
+        }
     }
-
 }
diff --git a/java/tests/SampleTest/src/com/android/rs/sample/SampleRSActivity.java b/java/tests/SampleTest/src/com/android/rs/sample/SampleRSActivity.java
index dd4a98a..d86dfed 100644
--- a/java/tests/SampleTest/src/com/android/rs/sample/SampleRSActivity.java
+++ b/java/tests/SampleTest/src/com/android/rs/sample/SampleRSActivity.java
@@ -66,7 +66,6 @@ public class SampleRSActivity extends Activity {
         }
 
         public boolean onSurfaceTextureDestroyed(SurfaceTexture surface) {
-            mOutPixelsAllocation.setSurface(null);
             return true;
         }
     }
@@ -102,11 +101,13 @@ public class SampleRSActivity extends Activity {
         mRS = RenderScript.create(this);
         mTwoByTwoAlloc = Allocation.createFromBitmap(mRS, mBitmapTwoByTwo,
                                                           Allocation.MipmapControl.MIPMAP_NONE,
-                                                          Allocation.USAGE_SCRIPT);
+                                                          Allocation.USAGE_SCRIPT |
+                                                          Allocation.USAGE_GRAPHICS_TEXTURE);
 
         mCityAlloc = Allocation.createFromBitmap(mRS, mBitmapCity,
                                                           Allocation.MipmapControl.MIPMAP_NONE,
-                                                          Allocation.USAGE_SCRIPT);
+                                                          Allocation.USAGE_SCRIPT |
+                                                          Allocation.USAGE_GRAPHICS_TEXTURE);
 
         Type.Builder b = new Type.Builder(mRS, Element.RGBA_8888(mRS));
 
diff --git a/java/tests/SampleTest/src/com/android/rs/sample/sample.rs b/java/tests/SampleTest/src/com/android/rs/sample/sample.rs
index e2bf43d..8b0ad23 100644
--- a/java/tests/SampleTest/src/com/android/rs/sample/sample.rs
+++ b/java/tests/SampleTest/src/com/android/rs/sample/sample.rs
@@ -37,7 +37,7 @@ void root(uchar4 *out, uint32_t x, uint32_t y) {
     uv.x = (float)x / destX;
     uv.y = (float)y / destY;
 
-    out->xyz = convert_uchar3(rsSample(sourceAlloc, allocSampler, uv*2.0f).xyz);
+    out->xyz = convert_uchar3(rsSample(sourceAlloc, allocSampler, uv*2.0f).xyz*255.0f);
     out->w = 0xff;
 }
 
-- 
1.7.4.1

